lib/facter/resolvers/freebsd/freebsd_version_resolver.rb

```diff
@@ -14,20 +14,22 @@ def post_resolve(fact_name)
           end
 
           def freebsd_version_system_call(fact_name)
-            output = Facter::Core::Execution.execute('/bin/freebsd-version -kru', logger: log)
-            return if output.empty?
+            output = Facter::Core::Execution.execute('/bin/freebsd-version -k', logger: log)
 
-            build_fact_list(output)
+            @fact_list[:installed_kernel] = output.strip unless output.empty?
+
+            output = Facter::Core::Execution.execute('/bin/freebsd-version -ru', logger: log)
+
+            build_fact_list(output) unless output.empty?
 
             @fact_list[fact_name]
           end
 
           def build_fact_list(output)
             freebsd_version_results = output.split("\n")
 
-            @fact_list[:installed_kernel]   = freebsd_version_results[0].strip
```

Comment:
why is this fact gone now?

Reply:
It's not gone, it's now collected on line 19.

This `build_fact_list` function is a bit misleading since it only does 2/3 of the job… Maybe moving the code to the calling function directly would be less confusing?

Reply:
`build_some_facts_list` 😅